cmd/dev.go

```diff
@@ -18,212 +12,104 @@ import (
 	Use:     "run",
 	Aliases: []string{"dev"},
 	Args:    cobra.NoArgs,
 	Short:   "Run the development server",
 	Run: func(cmd *cobra.Command, args []string) {
 		log := env.NewLogger(cmd)
-		sdkEventsFile := "events.log"
 		dir := project.ResolveProjectDir(log, cmd)
 		_, appUrl := getURLs(log)
 		websocketUrl := viper.GetString("overrides.websocket_url")
 		websocketId, _ := cmd.Flags().GetString("websocket-id")
 		apiKey, _ := util.EnsureLoggedIn()
 		theproject := project.EnsureProject(cmd)
 
+		// get project from api
 		project, err := theproject.Project.GetProject(log, theproject.APIURL, apiKey)
 		if err != nil {
-			log.Fatal("failed to get project: %s", err)
+			errsystem.New(errsystem.ErrInvalidConfiguration, err, errsystem.WithContextMessage(fmt.Sprintf("Failed to get project: %s", err))).ShowErrorAndExit()
 		}
-
 		orgId := project.OrgId
 
-		if _, err := os.Stat(sdkEventsFile); err == nil {
-			if err := os.Remove(sdkEventsFile); err != nil {
-				log.Trace("failed to delete sdkEventsFile: %s", err)
-			}
-		}
-
-		// get 6 random characters
+		// need to fixs this!!!!!!!
 		if websocketId == "" {
 			websocketId = uuid.New().String()[:6]
 		}
 
-		liveDevConnection, err := dev.NewLiveDevConnection(log, sdkEventsFile, websocketId, websocketUrl, apiKey)
+		liveDevConnection, err := dev.NewLiveDevConnection(log, websocketId, websocketUrl, apiKey, theproject)
 		if err != nil {
 			log.Fatal("failed to create live dev connection: %s", err)
 		}
 		defer liveDevConnection.Close()
-		devUrl := liveDevConnection.WebURL(appUrl)
-		log.Info("development server at url: %s", devUrl)
 
-		// Display local interaction instructions
-		displayLocalInstructions(theproject.Project.Development.Port, theproject.Project.Agents)
+		projectServerCmd, err := dev.CreateRunProjectCmd(log, theproject, liveDevConnection, dir, orgId)
+		if err != nil {
+			errsystem.New(errsystem.ErrInvalidConfiguration, err, errsystem.WithContextMessage("Failed to run project")).ShowErrorAndExit()
+		}
 
-		if err := browser.OpenURL(devUrl); err != nil {
-			log.Fatal("failed to open browser: %s", err)
+		build := func() {
+			fmt.Println(tui.Text(fmt.Sprintf("🔨 Building project...")))
+			started := time.Now()
+			if err := bundler.Bundle(bundler.BundleContext{
+				Context:    context.Background(),
+				Logger:     log,
+				ProjectDir: dir,
+				Production: false,
+			}); err != nil {
+				errsystem.New(errsystem.ErrInvalidConfiguration, err, errsystem.WithContextMessage(fmt.Sprintf("Failed to bundle project: %s", err))).ShowErrorAndExit()
+			}
+			fmt.Printf("✨ Built in %s\n", time.Since(started).Round(time.Millisecond))
 		}
-		logger := logger.NewMultiLogger(log, logger.NewJSONLoggerWithSink(liveDevConnection, logger.LevelInfo))
 
-		projectServerCmd, err := dev.RunProject(log, theproject, liveDevConnection, dir, orgId)
+		// Initial build
+		build()
+
+		// Watch for changes
+		watcher, err := dev.NewWatcher(dir, []string{"src/**"}, func(path string) {
```

Comment:
remember to get this information from the project Development since it's dependent on the language / framework